src/compiler/AST/Common/Type.elm

```diff
@@ -16,6 +16,7 @@ type Type
     | Char
     | String
     | Bool
+    | List (List Type)
```

Comment:
I think it has to be `List Type` (like in real Elm it's not `[Int, Float, String]` but `List Int`)